CoreScriptsRoot/Modules/BackpackScript.lua

```diff
@@ -173,6 +189,55 @@ local function AdjustHotbarFrames()
 	OpenInventoryButton.Visible = not inventoryOpen and visualTotal >= 1
 end
 
+local function AdjustInventoryFrames()
+	local lowestPoint = 0
+	for i = NumberOfHotbarSlots + 1, #Slots do
+		local slot = Slots[i]
+		slot:Reposition()
+		slot.Frame.Visible = (slot.Tool ~= nil)
+		lowestPoint = math.max(lowestPoint, slot.Frame.Position.Y.Offset + slot.Frame.Size.Y.Offset)
+	end
+	ScrollingFrame.CanvasSize = UDim2.new(0, 0, 0, lowestPoint + ICON_BUFFER)
+end
+
+local function UpdateBackpackLayout()
+	HotbarFrame.Size = UDim2.new(0, ICON_BUFFER + (NumberOfHotbarSlots * (ICON_SIZE + ICON_BUFFER)), 0, ICON_BUFFER + ICON_SIZE + ICON_BUFFER)
+	HotbarFrame.Position = UDim2.new(0.5, -HotbarFrame.Size.X.Offset / 2, 1, -HotbarFrame.Size.Y.Offset)
+	InventoryFrame.Size = UDim2.new(0, HotbarFrame.Size.X.Offset, 0, (HotbarFrame.Size.Y.Offset * NumberOfInventoryRows) + INVENTORY_HEADER_SIZE + (IsVR and 2*INVENTORY_ARROWS_BUFFER_VR or 0))
+	InventoryFrame.Position = UDim2.new(0.5, -InventoryFrame.Size.X.Offset / 2, 1, HotbarFrame.Position.Y.Offset - InventoryFrame.Size.Y.Offset)
+
+	ScrollingFrame.Size = UDim2.new(1, ScrollingFrame.ScrollBarThickness + 1, 1, -INVENTORY_HEADER_SIZE - (IsVR and 2*INVENTORY_ARROWS_BUFFER_VR or 0))
+	ScrollingFrame.Position = UDim2.new(0, 0, 0, INVENTORY_HEADER_SIZE + (IsVR and INVENTORY_ARROWS_BUFFER_VR or 0))
+	AdjustHotbarFrames()
+	AdjustInventoryFrames()
+end
+
+local function SetNumberOfHotbarSlots(numSlots)
```

Comment:
We already have to support the mid-game enabling of the VR headset because `UserInputService.VREnabled` is not true when the game launches for a considerable duration.
I think it is a good idea to make it ready to disable VR headset in the scenario we will need to switch between the two.